### ISSUE TITLE
get resource pool name under VM

### ISSUE THREAD

**Issue by ChaoJJ** [vmware/govmomi#2078]
I'm currently trying to switch from using the govc binary to govmomi library and can't quite get my head around how to get the resource pool name (not the id) for every one of my vms.
As a heads up, I've no much prior experience with vmware or vms in general, so please be patient.
The way it works right now with govc is that I get a list of all resource pool paths over `govc find -type resourcePool .`, later on I use the result in a new find request for VirtualMachines specifying the resource pool path.
I extract the pool name from the path.

Can you do the same with govmomi library? Whenever I try to get the name, I only get the ID _resgroup-xx_.

I've found a way to get all the resource pool paths with govmomi using the find package and calling the `ResourcePoolListAll` function and returning the `InventoryPath`
I know I can set the default datacenter when I use the find package, however I've not found a way to do the same for resource pool paths like I can with govc. 
Is it possible and is there an even better approach?

**Comment by dougm** [comment]
@ChaoJJ added an example in #2079 , can you review?

**Comment by ChaoJJ** [comment]
@dougm Thanks doug, the `ref, err := finder.ObjectReference(ctx, pool.Reference())` along with `fmt.Println(ref.(*object.ResourcePool).InventoryPath)` was what I needed. 
For some reason `name, err := pool.ObjectName(ctx)` only returns `"Resources"`, what I need however is `"/xx/Resources"`. 
Is the name property of a resourcePool just "Resources"?

However (*object.ResourcePool).InventoryPath) works fine, I can extract the name just like I did with govc. Thanks again.

Normally I'd close the ticket now, since my issue has been resolved, but I'd like to know what the name property is supposed to return of a resourcePool. I didn't mention that at the start, but I came upon the Name and the ObjectName attribute, but dismissed them as I only got returned `"Resources"` or an empty string.
